coverport/coverage_server.py

```
@@ -0,0 +1,54 @@
+    def do_GET(self):
+        if self.path == "/coverage":
+            try:
+                import coverage
+
+                cov = coverage.Coverage.current()
+                if cov:
+                    import tempfile
+
+                    with tempfile.NamedTemporaryFile(mode="w", suffix=".lcov", delete=False) as f:
+                        cov.lcov_report(outfile=f.name)
+                        data = open(f.name).read()
```

Ruff (0.15.12)

[warning] Use a context manager for opening files

(SIM115)